File: leetcode/Python/island-perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        sol, m ,n = 0, len(grid), len(grid[0])
        for i in range(m):
            for j in range(n):
                if grid[i][j]==1:
                    sol += 4
                    if j>0 and grid[i][j-1]==1:
                        sol -= 2
                    if i>0 and grid[i-1][j]==1:
                        sol -= 2
        return sol

File: leetcode/C++/island-perimeter.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int sol = 0, m = grid.size(), n=grid[0].size();
        for(int i=0; i<m; ++i){
            for(int j=0; j<n; ++j){
                if(grid[i][j]) {
                    sol+=4;
                    if(j && grid[i][j-1]) sol-=2;
                    if(i && grid[i-1][j]) sol-=2;
                }
            }
        }
        return sol;
    }
};